cmd/launcher/desktop.go

```diff
@@ -186,6 +193,22 @@ func monitorParentProcess(logger log.Logger) {
 	}
 }
 
+func parentProcessId() (int, error) {
+	if runtime.GOOS != "darwin" {
+		return os.Getppid(), nil
+	}
+
+	// On macOS, we run desktop as `launchctl asuser <UID> sudo --preserve-env -u <username> launcher desktop`, which
+	// means the parent process of the current process is the sudo process. We need to monitor one level above that
+	// to be monitoring the launcher root process.
+	sudoProcess, err := ps.FindProcess(os.Getppid())
```

Comment:
I think this will be racy around updates.

Can we pass the parent process id in as a command line arg or env?

Reply:
I am in favor of passing the pid as en env var. However, I just want to call out that this really is a back up plan for when root launcher crashes or gets manually killed. During a normal shut down like an auto update, desktop is shut down via a desktop server http cmd.

Reply:
Yup, will update 👍